### PR TITLE
Use inclusive language [removing sanity check]

### PR DESCRIPTION
**Pull request by chewselene**
Remove ableist language.

### CODE REVIEW

File: graphql_compiler/compiler/ir_validation.py [hunk 1]
```diff
@@ -255,7 +255,7 @@ def _sanity_check_every_location_is_marked(ir_blocks: List[BasicBlock]) -> None:
             mark_location_blocks_count = 0
 
 
-def _sanity_check_coerce_type_outside_of_fold(ir_blocks: List[BasicBlock]):
+def _validate_coerce_type_outside_of_fold(ir_blocks: List[BasicBlock]):
```
Comment (bojanserafimov) [inline graphql_compiler/compiler/ir_validation.py:258]:
```suggestion
def _validate_coerce_type_outside_of_fold(ir_blocks: List[BasicBlock]) -> None:
```

**Review by branen — COMMENTED** [review]
Instead of `self_consistency_check` and `self_consistency_checks`, what about `assert` and `assertions`?  They describe what's actually happening, are unambiguous to the reader, and are much easier to read, type, and say.

**Comment by chewselene** [comment]
> Instead of `self_consistency_check` and `self_consistency_checks`, what about `assert` and `assertions`? They describe what's actually happening, are unambiguous to the reader, and are much easier to read, type, and say.

I originally had this as validate/validation! gonna loop in @obi1kenobi who said:
` would it be okay to use self-consistency check instead of validate when possible for this? there’s a subtle distinction, but compiler IR can be locally valid (all real blocks with valid arguments etc.) but self-inconsistent (invariants that span multiple blocks are not upheld) so for example, if there’s a Traverse without a matching Backtrack, or a Traverse to a location that is never marked with MarkLocation`

I think assert/assertion likely has a similar issue to validate/validation, but Branen also pointed out that all these functions do raise assertion errors so maybe it is appropriate?

**Review by obi1kenobi — APPROVED** [review]
Looks fine to me!

**Pull request merged**